src/main/java/com/google/devtools/build/lib/analysis/starlark/Args.java

```diff
@@ -454,6 +475,13 @@ private void addVectorArg(
         if (expandDirectories) {
           scanForDirectories(starlarkList);
         }
+        // Labels referencing targets in the main repo are stringified as //pkg:name and thus
+        // don't require a RepositoryMapping. If a map_each function is provided, default
+        // stringification via Label#toString() is not used.
+        mayStringifyExternalLabel |=
+            (mapEach != null)
+                && starlarkList.stream()
```

Comment:
If expand_directories is true (the default) we already iterate over this list. Can you hook into that so that we only have one iteration? Can still do this if expand_directories is false.